modules/fee-market/src/lib.rs

```diff
@@ -346,7 +379,19 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	/// - The enrolled relayer wants to update fee or order capacity.
 	/// - The enrolled relayer wants to cancel enrollment.
 	/// - The order didn't confirm in-time, slash occurred.
-	pub(crate) fn update_market() {
+	pub(crate) fn update_market<F>(
+		f: F,
+		has_event: Option<Event<T, I>>,
+	) -> DispatchResultWithPostInfo
+	where
+		F: FnOnce() -> DispatchResultWithPostInfo,
```

Comment:
It looks like all the `F` return `Ok(().into())`.
Do we really need this?

Reply:
Good question. Sometimes some checks need to be done in the `F`, see `update_locked_collateral(...)'s F`, so it is necessary to me now.

Reply:
But I don't see `F` threw out any error.

Reply:
Any specific reason to use `DispatchResultWithPostInfo` rather then `DispatchResult`?

Reply:
`ensure!(new_collateral >= orders_locked_collateral,<Error<T, I>>::StillHasOrdersNotConfirmed);` in `update_locked_collateral(...)'s F`, then `f()?` in this function.

Reply:
> Any specific reason to use DispatchResultWithPostInfo rather than DispatchResult?

What is the benefit after changing to `DispatchResult`?